src/StructTypes.jl

```diff
@@ -9,8 +9,15 @@ StructType(x::T) where {T} = StructType(T)
 
 "Default `StructTypes.StructType` for types that don't have a `StructType` defined; this ensures objects must have an explicit `StructType` to avoid unanticipated issues"
 struct NoStructType <: StructType end
+struct SingletonType <: StructType end
 
-StructType(::Type{T}) where {T} = NoStructType()
+function StructType(::Type{T}) where {T}
```

Comment:
I lean towards implementing https://github.com/JuliaData/StructTypes.jl/issues/61 before going ahead with this; in this case, we're saying that *all* singleton types automatically have a struct type, even though that hasn't been the policy. If you don't mind taking a stab at the implementation, it should be as simple as defining something like:
```julia
StructType(::Type{T}) where {T} = Struct()
```

once we have that implemented and merged, we can then update this PR to do basically the same that you have here, except return `Struct` instead of `NoStructType`.

Reply:
This change has an interesting side effect, since functions are singletons:
```
julia> StructTypes.StructType(print)
StructTypes.SingletonType()
```

Reply:
Based on this comment (https://github.com/JuliaData/StructTypes.jl/pull/67#issuecomment-983822907)  I think this behaviour is actually correct except for calling `StructTypes.StructType(print)` instead of `StructTypes.StructType(typeof(print))`.  I think it may be a concern for my follow-up MR in JSON3 though